src/Gridify/Builder/BaseQueryBuilder.cs

```diff
@@ -6,6 +6,11 @@
 
 namespace Gridify.Builder;
 
+internal static class BaseQueryBuilder
+{
+   internal static MemberNullPropogationVisitor MemberNullPropagatorVisitor { get; } = new();
```

Comment:
Is there any specific reason that this is a static object? I think for consistency, it's better to instantiate if needed, unless you have reason to keep it static.